hcl2/doc.go

```diff
@@ -0,0 +1,5 @@
+/*
+Package hcl2 provides helper functions for decoding interfaces of a single config value or
+a whole config struct into a cty.Value.
```

Comment:
Here, I'd avoid naming this pkg `hcl2`, to maybe name it to something that does not exist, also in the header we could say :

// Package ... provides helper functions for passing or getting hcl2 types to and from a Packer plugin.

Also, `hcl2.HCL2ValueFromConfigValue` and `hcl2.HCL2ValueFromConfig` will stutter.

In general, I'd advise on maybe thinking about the way the godocs api will look first here because that's what sdk using people will check out first.

Reply:
is hcl2shim a good name? the same we use in the packer core 

> Also, hcl2.HCL2ValueFromConfigValue and hcl2.HCL2ValueFromConfig will stutter.
What do you mean by that? 🤔

Reply:
Yes ! `hcl2shim` sounds like a good name ! Also, before seeing your message, after some thinking and because hcl is specific to each projects (terraform/consul/nomad/etc.) I though of maybe naming it `pkrhcl2helper` because its a helper for hcl2 that is specific to packer, I really like it but the name is kinda long I'd like to let you decide there.

Stutter means that the name of the function contains/repeats the name of the package, ex `hcl2.HCL2ValueFromConfig` and we could change it to: `hcl2.ValueFromConfig`, I reuse the wording from the go community on good practices.

Reply:
Nice, didn't know the term. Thanks! 
It makes sense everything you're saying. I'll make the changes here!